addons/sourcemod/scripting/superzombiefortress.sp

```diff
@@ -1404,633 +1331,563 @@ public Action event_RoundStart(Event event, const char[] name, bool dontBroadcas
+				if (fHighest < g_flDamageDealtAgainstTank[i])
 				{
-					fHighest = g_fDamageDealtAgainstTank[i];
+					fHighest = g_flDamageDealtAgainstTank[i];
 					iWinner = i;
 				}
-
+				
 				AddMorale(i, 20);
 			}
 		}
-
+		
 		if (fHighest > 0.0)
 		{
 			SetHudTextParams(-1.0, 0.3, 8.0, 200, 255, 200, 128, 1);
-
+			
 			for (int i = 1; i <= MaxClients; i++)
-			{
 				if (IsValidClient(i))
-				{
-					ShowHudText(i, 5, "The Tank '%N' has died\nMost damage: %N (%d)", victim, iWinner, RoundFloat(fHighest));
-				}
-			}
+					ShowHudText(i, 5, "The Tank '%N' has died\nMost damage: %N (%d)", iVictim, iWinner, RoundFloat(fHighest));
 		}
-
-		if (g_iDamageDealtLife[victim] <= 50 && g_iDamageTakenLife[victim] <= 150 && !g_bTankRefreshed)
+		
+		if (g_iDamageDealtLife[iVictim] <= 50 && g_iDamageTakenLife[iVictim] <= 150 && !g_bTankRefreshed)
 		{
 			g_bTankRefreshed = true;
-			g_iSpecialInfected[victim] = INFECTED_NONE;
+			g_nInfected[iVictim] = Infected_None;
 			ZombieTank();
 		}
 		
 		Call_StartForward(g_hForwardTankDeath);
-		Call_PushCell(victim);
+		Call_PushCell(iVictim);
 		Call_PushCell(iWinner);
 		Call_PushCell(RoundFloat(fHighest));
 		Call_Finish();
 	}
-
-	g_iEyelanderHead[victim] = 0;
-	g_iMaxHealth[victim] = -1;
-	g_ShouldBacteriaPlay[victim] = true;
-	g_bReplaceRageWithSpecialInfectedSpawn[victim] = false;
-	int g_iSpecialInfectedIndex = g_iSpecialInfected[victim];
-	g_iSpecialInfected[victim] = INFECTED_NONE;
-
-	// Handle zombie death logic, all round states.
-	if (IsValidZombie(victim))
+	
+	g_iEyelanderHead[iVictim] = 0;
+	g_iMaxHealth[iVictim] = -1;
+	g_ShouldBacteriaPlay[iVictim] = true;
+	g_bReplaceRageWithSpecialInfectedSpawn[iVictim] = false;
+	
+	Infected g_nInfectedIndex = g_nInfected[iVictim];
+	g_nInfected[iVictim] = Infected_None;
+	
+	//Handle zombie death logic, all round states.
+	if (IsValidZombie(iVictim))
 	{
-		// 10%
-		if (IsValidSurvivor(killers[0]) && !GetRandomInt(0, 9) && g_bRoundActive == true)
-		{
-			g_bSpawnAsSpecialInfected[victim] = true;
-		}
+		//10%
+		if (IsValidSurvivor(iKillers[0]) && !GetRandomInt(0, 9) && g_nRoundState == SZFRoundState_Active)
+			g_bSpawnAsSpecialInfected[iVictim] = true;
 		
-		// boomer
-		if (g_iSpecialInfectedIndex == INFECTED_BOOMER)
-		{
-			DoBoomerExplosion(victim, 400.0);
-		}
+		//Boomer
+		if (g_nInfectedIndex == Infected_Boomer)
+			DoBoomerExplosion(iVictim, 400.0);
 		
-		// set special infected state
-		if (g_iNextSpecialInfected[victim] != INFECTED_NONE)
+		//Set special infected state
+		if (g_nNextInfected[iVictim] != Infected_None)
 		{
-			if (victim != g_iZombieTank) g_iSpecialInfected[victim] = g_iNextSpecialInfected[victim];
-			g_iNextSpecialInfected[victim] = INFECTED_NONE;
+			if (iVictim != g_iZombieTank) g_nInfected[iVictim] = g_nNextInfected[iVictim];
+			g_nNextInfected[iVictim] = Infected_None;
 		}
 		
-		// Remove dropped ammopacks from zombies.
+		//Remove dropped ammopacks from zombies.
 		int index = -1;
```

Comment:
should be `iIndex`

Reply:
`iEntity` fits better, i would've think of `iIndex` as weapon's item index defs